sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -240,19 +242,41 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
 
   /**
    * Update DisbursementSchedule with documentNumber
-   * @param documentNumber document Number
+   * The update to Application and Disbursement schedule happens in single transaction.
+   * @param disbursementScheduleId document Number
    * @returns the result of update.
    */
-  async updateDisbursementScheduleDocumentNumber(
+  async updateDisbursementAndApplicationCOEApproval(
+    disbursementScheduleId: number,
+    userId: number,
     applicationId: number,
-  ): Promise<UpdateResult> {
+    applicationStatus: ApplicationStatus,
+  ): Promise<void> {
     const documentNumber = await this.getNextDocumentNumber();
-    return this.repo
-      .createQueryBuilder()
-      .update(DisbursementSchedule)
-      .set({ documentNumber: documentNumber })
-      .where("application.id = :applicationId", { applicationId })
-      .execute();
+    this.connection.transaction(async (transactionalEntityManager) => {
+      await transactionalEntityManager
+        .getRepository(DisbursementSchedule)
+        .createQueryBuilder()
+        .update(DisbursementSchedule)
+        .set({
+          documentNumber: documentNumber,
+          coeStatus: COEStatus.completed,
+          coeUpdatedBy: { id: userId },
+          coeUpdatedAt: new Date(),
```

Comment:
shouldn't it be UTC date? if so, we do have a function called `getUTCNow()`

Reply:
@ann-aot, we can talk about it. Our API and DB and Openshift are all running on UTC. _new Date()_ and getUTCCNow() will return exactly the same value on this scenario. Should we continue to use the getUTCCNow when the  _new Date()_ is enought?